From a701886b5cd99775b38234ac7156a0b861dc25e5 Mon Sep 17 00:00:00 2001
From: Alexander Grund <Flamefire@users.noreply.github.com>
Date: Tue, 1 Apr 2025 19:10:14 +0200
Subject: Update minimum CMake version

With the release of CMake 4 CMake < 3.5 is no longer supported and at least 3.10 is deprecated.

Update to 3.16 as the minimum and remove some of the workarounds and policies.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e44380c..8108579 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,21 +1,4 @@
-cmake_minimum_required(VERSION 3.1..3.18)
-
-# Policies
-
-# Include file check macros honor CMAKE_REQUIRED_LIBRARIES, CMake >= 3.12
-if(POLICY CMP0075)
-  cmake_policy(SET CMP0075 NEW)
-endif()
-
-# MSVC runtime library flags are selected by an abstraction, CMake >= 3.15
-# This policy still need to be set even with cmake_minimum_required() command above.
-if(POLICY CMP0091)
-  if(DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
-    cmake_policy(SET CMP0091 NEW)
-  else()
-    cmake_policy(SET CMP0091 OLD)
-  endif()
-endif()
+cmake_minimum_required(VERSION 3.16..3.20)
 
 project(libsamplerate VERSION 0.2.2 LANGUAGES C)
 
@@ -137,11 +120,4 @@ endif()
 
 # Packaging support
 
-# See https://cmake.org/cmake/help/v3.12/release/3.12.html#cpack
-if(CMAKE_VERSION VERSION_LESS 3.12)
-  set(CPACK_PACKAGE_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
-  set(CPACK_PACKAGE_VERSION_MINOR ${PROJECT_VERSION_MINOR})
-  set(CPACK_PACKAGE_VERSION_PATCH ${PROJECT_VERSION_PATCH})
-endif()
-
 include(CPack)
-- 
2.50.1

